### PR TITLE
community: add SearchResourceEncryptionKey support

### PR DESCRIPTION
**Pull request by seb-jul**
**Description**

- Added support for SearchResourceEncryptionKey from azure.search.documents.indexes.models. This enhancement allows for encryption key management within Azure Search resources.

- Added mock tests

- [x] Unit tests pass locally

### CODE REVIEW

**Comment by seb-jul** [comment]
Hello @baskaryan could you please review this PR ?

**Comment by seb-jul** [comment]
Hello @efriis could you please review this PR ?

**Comment by seb-jul** [comment]
Hello @baskaryan and @efriis could you please review this PR ?

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -200,12 +203,23 @@ def fmt_err(x: str) -> str:
             # don't use semantic search
             semantic_search = None
 
+        # Create the encryption key with the configuration
+        if search_resource_encryption_key is not None:
+            search_resource_encryption_key = SearchResourceEncryptionKey(
+                key_name="default_key_name",
+                key_version="default_key_version",
+                vault_uri="https://default-keyvault-name.vault.azure.net",
+            )
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/azuresearch.py:212]: why are we overwriting the key if its *not* None?

Comment (baskaryan) [reply libs/community/langchain_community/vectorstores/azuresearch.py:212]: bump on this @seb-jul, are you still working on this pr?

**Pull request closed without merging**